src/OracleService/Protocols/OracleHttpsProtocol.cs

```diff
@@ -82,6 +82,8 @@ public async Task<(OracleResponseCode, string)> ProcessAsync(Uri uri, Cancellati
                 return (OracleResponseCode.Error, message.StatusCode.ToString());
             if (!Settings.Default.AllowedContentTypes.Contains(message.Content.Headers.ContentType.MediaType))
                 return (OracleResponseCode.ContentTypeNotSupported, null);
+            if (!message.Content.Headers.ContentLength.HasValue || message.Content.Headers.ContentLength > OracleResponse.MaxResultSize)
```

Comment:
The `!message.Content.Headers.ContentLength.HasValue` part of it may affect legitimate responses that fit the limit, but don't have the header for whatever reason (dynamically generated content, for example).

Reply:
We can require this header

Reply:
That'd be an artificial limitation to me. It shouldn't be hard to handle `!message.Content.Headers.ContentLength.HasValue` case as well, NeoGo handles both cases easily.